components/camel-cxf/src/main/java/org/apache/camel/component/cxf/WSDLServiceFactoryBean.java

```diff
@@ -102,7 +102,7 @@ protected void buildServiceFromWSDL(String url) {
         super.buildServiceFromWSDL(url);
     }
     public Endpoint createEndpoint(EndpointInfo ei) throws EndpointException {
-        Endpoint ep = new EndpointImpl(getBus(), getService(), ei);
+        Endpoint ep = new JaxWsEndpointImpl(getBus(), getService(), ei);
```

Comment:
this may be a breaking change.

Reply:
That class extends EndpointImpl and I think its fine to allow changes like this in a new future Camel release.